### DISCUSSION TITLE
How to upgrade Celery from AMQP to RPC?

### DISCUSSION THREAD

**Discussion by rationa1** [celery/celery#7629] [Q&A]
How do you upgrade a Celery 4 configuration, that uses the AMQP backend, to Celery 5, which has dropped this support and now requires you to use the RPC backend?

After upgrading to Celery 5.1.2, I received the error:

ModuleNotFoundError: No module named 'celery.backends.amqp'
which I expected. However, it's unclear how the rpc backend replaces this.

My CELERY_BROKER_URL looks like amqp://guest:guest@127.0.0.1:5672/myvhost, which accesses RabbitMQ.

I tried changing this to rpc://guest:guest@127.0.0.1:5672/myvhost, and I changed my CELERY_RESULT_BACKEND to rpc://, but that just gives me a different error:

KeyError: 'No such transport: rpc'
While there's a ton of [documentation](https://docs.celeryproject.org/en/stable/internals/reference/celery.backends.rpc.html), it's unclear what I'm doing wrong. How do you upgrade Celery from amqp to rpc?

[a comment by an automated account is not shown]

**Comment by emperinter** [comment]
I get the same error as follows on celery 5.1.2, and the solution i can found is down to 4.4.6

- my config 

```python
broker_url = 'amqp://guest@localhost//'
result_backend = 'amqp://guest@localhost//'
```

- error 

```shell
ModuleNotFoundError: No module named 'celery.backends.amqp'
```
- the documation in offical website as follows:

![图片](https://user-images.githubusercontent.com/20177836/175914480-240d3d2c-948a-463c-9d3e-6d3dab16760f.png)

**Comment by sankhakarfa** [comment]
Hi,

I had the same issue in celery==5.2.7
I looked into this documentation https://docs.celeryq.dev/en/v5.2.7/getting-started/first-steps-with-celery.html#keeping-results
and fixed it

Use the following:
 
`backend='rpc://', `
`broker='pyamqp://'`

And it should work.

**Comment by emperinter** [comment]
@sankhakarfa It works for me,Thx!

```python
backend='rpc://guest@localhost//'
broker='pyamqp://guest@localhost//'
```
